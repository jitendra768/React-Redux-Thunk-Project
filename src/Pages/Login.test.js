/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react";
import Login from "./Login";
import React from "react";
test("LoginForm submits correct data", () => {
  const handleSubmitForm = jest.fn();
  const { asFragment }=render(<Login onSubmit={handleSubmitForm} />);

  // Fill out the form
  fireEvent.change(screen.getByPlaceholderText("Email"), {
    target: { value: "test@example.com" },
  });
  fireEvent.change(screen.getByPlaceholderText("Password"), {
    target: { value: "password123" },
  });

  // Submit the form
  fireEvent.click(screen.getByText("Sign in"));

  // Check if the form was submitted with the correct data
  expect(handleSubmitForm).toHaveBeenCalledWith({
    email: "test@example.com",
    password: "password123",
  });
  expect(asFragment()).toMatchSnapshot();
});
