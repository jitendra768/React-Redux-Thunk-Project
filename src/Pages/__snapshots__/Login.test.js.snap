// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm submits correct data 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8"
  >
    <div
      class="sm:mx-auto sm:w-full sm:max-w-sm"
    >
      <img
        alt="Your Company"
        class="mx-auto h-10 w-auto"
        src="https://tailwindui.com/plus/img/logos/mark.svg?color=indigo&shade=600"
      />
      <h2
        class="mt-10 text-center text-2xl/9 font-bold tracking-tight text-gray-900"
      >
        Sign in to your account
      </h2>
    </div>
    <div
      class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
    >
      <form
        action="#"
        class="space-y-6"
        method="POST"
      >
        <div>
          <label
            class="block text-sm/6 font-medium text-gray-900"
            for="email"
          >
            Email address
          </label>
          <div
            class="mt-2"
          >
            <input
              autocomplete="email"
              class="block w-full rounded-md bg-white px-3 py-1.5 text-base text-gray-900 outline-1 -outline-offset-1 outline-gray-300 placeholder:text-gray-400 focus:outline-2 focus:-outline-offset-2 focus:outline-indigo-600 sm:text-sm/6"
              id="email"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value="test@example.com"
            />
          </div>
        </div>
        <div>
          <div
            class="flex items-center justify-between"
          >
            <label
              class="block text-sm/6 font-medium text-gray-900"
              for="password"
            >
              Password
            </label>
            <div
              class="text-sm"
            >
              <a
                class="font-semibold text-indigo-600 hover:text-indigo-500"
                href="#"
              >
                Forgot password?
              </a>
            </div>
          </div>
          <div
            class="mt-2"
          >
            <input
              autocomplete="current-password"
              class="block w-full rounded-md bg-white px-3 py-1.5 text-base text-gray-900 outline-1 -outline-offset-1 outline-gray-300 placeholder:text-gray-400 focus:outline-2 focus:-outline-offset-2 focus:outline-indigo-600 sm:text-sm/6"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value="password123"
            />
          </div>
        </div>
        <div>
          <button
            class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm/6 font-semibold text-white shadow-xs hover:bg-indigo-500 focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </form>
      <p
        class="mt-10 text-center text-sm/6 text-gray-500"
      >
        Not a member? 
        <a
          class="font-semibold text-indigo-600 hover:text-indigo-500"
          href="#"
        >
          Start a 14 day free trial
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
